### DISCUSSION TITLE
node:sqlite in Meteor

### DISCUSSION THREAD

**Discussion by StorytellerCZ** [meteor/meteor#13234] [Feature Requests]
So with [Node 22.5](https://nodejs.org/en/blog/release/v22.5.0) we now have sqlite available to us without the need of any extra packages. I think it will take a bit to improve the speed of the official package and other things over third-party solutions, but I think we should be looking at this long term for two perspectives:

**1. Internal use of sqlite** 
Meteor uses sqlite internally. I think, once Meteor updates to Node 22, it should be migrated to use the Node version. Provided that it has all the functionality that is needed, which I think should be the case. This will remove one extra dependency to take care of.

**2. Usage in apps**
Given that this will now be part Node, it will be available to use in apps. I'm primarily opening this discussion to chat about what should that look like. At the very least it would be good to provide guidance, but I think we might even want to consider if we can go beyond that. I think it might be worth considering if we could do reactivity with sqlite like we do with MongoDB, probably with help of FerretDB or the like, or maybe sqlite could play another role in the ecosystem. At the very least I think it will be a good idea to provide some sort of guidance on this.

**Comment by jankapunkt** [comment]
Would it be isomorph to the sqlite in the browsers?

**Reply by StorytellerCZ** [reply]
We need to talk about that. I think the first thing we need to figure out would be the ideal use case.

**Reply by nachocodoner** [reply]
I think this is just for the server-side, you would still have to install the sqlite for browser.

**Reply by jankapunkt** [reply]
@nachocodoner you mean something like https://github.com/sql-js/sql.js/ ?

**Reply by nachocodoner** [reply]
Yes!, actually that one is the library for browser support. I know it is indirectly because when using capcitorjs sqlite plugin, it uses that one behind the scenes.

**Reply by StorytellerCZ** [reply]
I think the first step will be to migrate to it in the tool to reduce external dependency there.
If we can then make it isomorphic to client then I think we should look at the whole picture with Minimongo in mind as well. So this whole thing can go into something bigger or stay relatively small. I think provided that this is now in core of Node it might be worth it to explore this in detail.

**Reply by xet7** [reply]
There was audit for Standard for Public Code made to Meteor 2 based WeKan Open Source kanban.

Requirement: The codebase MUST be independent from any secret, undisclosed, proprietary or non-open licensed software or services for execution and understanding.

Notes and links: Depends upon a database with a non-open license, expect this shall change in the coming year.

I'm trying to find a way to use SQLite, or any other database that has open license.

https://wekan.github.io/standard-for-public-code/

**Reply by StorytellerCZ** [reply]
I'm not sure I follow the issue here. Is the issue MongoDB license?

**Reply by xet7** [reply]
@StorytellerCZ 

Yes. MongoDB SSPL license is not Open Source license.

https://en.wikipedia.org/wiki/Server_Side_Public_License

**Reply by jankapunkt** [reply]
@xet7 there is FerretDB with MondoDB compatibility maybe it works already? They announced a while ago to support Meteor

**Reply by xet7** [reply]
@jankapunkt 

Yes, previously FerretDB announced OpLog support:

https://forums.meteor.com/t/ferretdb-1-18-now-has-oplog-support-trying-replace-mongodb-6-x-with-ferretdb-postgresql-or-ferretdb-sqlite/61092

I will try FerretDB again. Previously I did get some errors:

https://github.com/FerretDB/FerretDB/issues/1752#issuecomment-1984464772
